libs/design/src/molecules/sidebar/animation/sidebar-animation.ts

```diff
@@ -8,34 +8,51 @@ import {
 } from '@angular/animations';
 
 const duration = '350ms';
-const sidebarAnimateRemainTransition = 'cubic-bezier(0.4, 0.0, 0.2, 1)';
 const sidebarAnimateInTransition = 'cubic-bezier(0.0, 0.0, 0.2, 1)';
 const sidebarAnimateOutTransition = 'cubic-bezier(0.4, 0.0, 1, 1)';
 
 export const daffSidebarAnimations: {
   readonly transformSidebar: AnimationTriggerMetadata;
   readonly transformContent: AnimationTriggerMetadata;
+  readonly backdropTrigger: AnimationTriggerMetadata;
 } = {
   transformSidebar: trigger('transformSidebar', [
     // We remove the `transform` here completely, rather than setting it to zero, because:
     // 1. 3d transforms causes text to appear blurry on IE and Edge.
     state('open', style({
       transform: 'none',
-      visibility: 'visible',
     })),
+    state('closed', style({
+      transform: 'translateX({{width}})',
+    }), { params: { width: '-300px' }}),
     transition('open => closed', animate(duration + ' ' + sidebarAnimateOutTransition)),
     transition('closed => open', animate(duration + ' ' + sidebarAnimateInTransition)),
   ]),
   transformContent: trigger('transformContent', [
+    // We remove the `transform` here completely, rather than setting it to zero, because:
+    // 1. 3d transforms causes text to appear blurry on IE and Edge.
     state('closed', style({
       transform: 'none',
     })),
+    state('open', style({
+      transform: 'translateX({{shift}})',
+    }), { params: { shift: '-300px' }}),
```

Comment:
why is this hardcoded to `-300px`?

Reply:
I agree, it should be dynamic given what the user wants their sidebar width to be.

Reply:
It is dynamic, I was required to set a default value, so I picked the one that corresponds with our sidebar width. I can add as a comment about synchronization.

Reply:
So if I add a class `.app-sidebar` to the sidebar component and set a custom width, this shift will be updated to that width as well?

Reply:
Yes! https://github.com/graycoreio/daffodil/pull/2191/files#diff-2c8ca20d6021f3f48fba4af999b1651b4db94821c7f1c96f11d8428fae8e6b06R84